pkg/gather/clusterconfig/clusterconfig_gatherer.go

```diff
@@ -0,0 +1,94 @@
+package clusterconfig
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/insights-operator/pkg/anonymization"
+	"github.com/openshift/insights-operator/pkg/gather/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// TODO: docs
+type Gatherer struct {
+	gatherKubeConfig        *rest.Config
+	gatherProtoKubeConfig   *rest.Config
+	metricsGatherKubeConfig *rest.Config
+	anonymizer              *anonymization.Anonymizer
+}
+
+type GathererFuncPtr = func(*Gatherer, context.Context) ([]record.Record, []error)
+
+var gatheringFunctions = map[string]common.GatheringFunction{
+	"pdbs":                              common.ImportantFunc((*Gatherer).GatherPodDisruptionBudgets),
```

Comment:
`(*Gatherer).GatherPodDisruptionBudgets` hmmm....it's a bit confusing (at least to me), but I guess I don't have any better idea.

Reply:
What do you mean?

Reply:
Well I know what is the purpose, but I don't know how to call this declaration :)  It's not type casting, is it? It looks like that I don't know Golang at all... :)

Reply:
Wouldn't be better to keep the gathering function eg ` GatherPodDisruptionBudgets(g *Gatherer, ctx context.Context)` as we have it right now?

Reply:
It's basically a pointer to a method. Well, there's no differences from the point of how it works, but I think it makes more sense to convert the functions to methods cuz they all use the same data from the gatherer.

Reply:
OK thanks. Yeah there's no difference. We don't use them like methods anyway. It's minor and up to you. It seemed to me more straightforward without the pointer.